### PR TITLE
Re-Build Lock File

### PR DESCRIPTION
**Pull request by arichiv**
In order to change package versions the package-lock should be re-built. This breaks a lot of things, so it's been split out as an independent change. #936

### CODE REVIEW

**Comment by arichiv** [comment]
Okay, so it looks like it has been a while since this was re-built. I'll try playing around with the package versions a bit, but at some point it's not worth putting off the work of just sitting on edge (especially for `@types/react`).

**Comment by arichiv** [comment]
@moog16 it's alive! The most interesting change here is locking `@types/react` and `react` at `16.7.X` due to [issues with `16.8.X`](https://github.com/material-components/material-components-web-react/issues/781).

**Comment by arichiv** [comment]
@moog16 I'm actually not updating them (much)! The locked-in versions were much newer than the requested version as `^16.0.0` asks for the latest `16.x` release while `~16.0.0` asks for the latest `16.0.x` release.

If you would prefer, I can lock in the older `react(-dom)` versions, but updating to match the `@types/*` packages already in use (by time of release) seemed to make sense.

### `react`
Did ask for `^16.4.2` and locked `16.4.2`.
Will ask for `~16.7.0` and locks `16.7.0`.

### `@types/react`
Did ask for `^16.4.4` and locked `16.7.17`.
Will ask for `~16.7.22` and locks `16.7.22`.

### `react-dom`
Did ask for `^16.4.2` and locked `16.4.2`.
Will ask for `~16.7.0` and locks `16.7.0`.

### `@types/react-dom`
Did ask for `^16.0.11` and locked `16.0.11`.
Will ask for `~16.0.11` and locks `16.0.11`.

**Comment by moog16** [comment]
@arichiv interesting...so you're saying we're "downgrading"? Why not just stick to the version we're on?

**Comment by arichiv** [comment]
@moog16 the table above indicates all packages will be locked at newer (or identical) versions. None have been downgraded.

What is changing is that the primary `package.json` (not requirements for sub-packages) is no longer indicating compatibility the repo can't support. As an example, in `@types/react` we indicate `^16.4.4` is supported, which means any and all versions through the current `16.8.20` work. This is a lie, as there are numerous type errors with the `16.8.x` branch. This hadn't been obvious before as the `package-lock.json` file was forcing npm to pull in `16.7.17`. With this change the `package.json` properly indicates that we support up through `16.7.x` and the version locked in is updated to `16.7.22`.

**Comment by moog16** [comment]
@arichiv gotcha - thanks for fixing those! Looks like there is a conflict with package-lock again. I'll approve once fixed and I can pull in the changes to take a look myself.

File: packages/chips/Chip.tsx [hunk 1]
```diff
@@ -70,8 +70,8 @@ export class Chip extends React.Component<ChipProps, ChipState> {
   };
 
   state = {
-    classList: new Set(),
-    leadingIconClassList: new Set(),
+    classList: new Set<string>(),
```
Comment (moog16) [inline packages/chips/Chip.tsx]: are these required? Isn't this inferred by ChipState?

Comment (arichiv) [reply packages/chips/Chip.tsx]: Turns out that they aren't anymore :-/

Comment (moog16) [reply packages/chips/Chip.tsx]: Ah can't we set the type of the object by doing:

`state: ChipState = {...}`

**Comment by arichiv** [comment]
This diff broke types due to an upgrade to the typescript package. https://github.com/material-components/material-components-web-react/pull/952 is the fix

**Pull request merged**